### ISSUE TITLE
Guest window doesn't follow guest on slide

### ISSUE THREAD

**Issue by CorySanin** [OpenRCT2/OpenRCT2#21487]
### Operating System

Artix Linux

### OpenRCT2 build

OpenRCT2, v0.4.8

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This bug is a graphical glitch or error

### Describe the issue

![guest-view-slide](https://github.com/OpenRCT2/OpenRCT2/assets/2897351/a1978cbd-e474-45cf-a069-3510ceb92fb1)

The viewport on the guest window goes blank when they enter a slide building. I see it on the latest develop appimage too. Which doesn't seem to say the commit hash its from? But it's 0.4.8 with network version 2.

### Steps to reproduce

1. Build a spiral slide
2. Select a guest that's about to ride
3. Watch the viewport

**Comment by ocalhoun6** [comment]
This has been a problem for a *long* time. I think I made an old bug report about it, but I can't find it at the moment.

Anyway, it's not truly blank -- when the guest window goes dark, it's actually centered *way* off in the void. (I'm guessing the coordinates are 0,0 or something like that?) You can see this if you set the main view to follow that guest -- when the guest is on the spiral slide, the main view will go way off into the void as well, but since it's bigger, you'll still be able to see the park in one corner of the screen.

Actually following the guest's circular path around the slide is probably too much to ask for ... but the view should be centered on the slide while the guest is on it.

**Comment by CorySanin** [comment]
> I think I made an old bug report about it

I searched open issues before posting, didn't find anything either.

I suspected that the view is focusing on 0,0, I think you might be right about that. Also using "follow on main viewport" is a good idea for debugging.

**Comment by duncanspumpkin** [comment]
Slides are a whole bunch of special case code so I'm not surprised it doesn't work.

**Comment by Harry-Hopkinson** [comment]
It seems to be related to these lines here.

https://github.com/OpenRCT2/OpenRCT2/blob/42844d1667bbbdc31825c27e68a9c6e8a8d6ce28/src/openrct2/interface/Viewport.cpp#L744C13-L754C14

Would it be as easy to add another if statement to check if it is a flat ride and then just center the view on the peep? Or is it more complicated than that?

**Comment by rik-smeets** [comment]
I can't reproduce this, the viewport just stays centered on the spiral slide here. The rendering mode doesn't seem to matter either. Only difference is that I'm running Windows.

**Comment by ocalhoun6** [comment]
I just tested and I'm still able to reproduce it... (Ubuntu 22.04)

Weird... Is this actually a Linux-specific bug? Doesn't seem like the sort of thing that would be affected by the host OS.

**Comment by rik-smeets** [comment]
Yeah I also wouldn't say so, but I don't see any other variable which might explain that.

**Comment by ocalhoun6** [comment]
I just tested in Windows in OpenGL, and it still happened for me.

**Comment by Harry-Hopkinson** [comment]
I cannot reproduce on MacOS for some reason.

**Comment by Gymnasiast** [comment]
I cannot reproduce this on Ubuntu 24.04. Testing with OpenGL and software rendering. Please check again.

**Comment by ocalhoun6** [comment]
Just tested on Windows OpenGL again, and it works properly for me now.

**Comment by CorySanin** [comment]
<img width="494" height="316" alt="Image" src="https://github.com/user-attachments/assets/0994f062-dd2e-4ad4-a45a-3df8a7b00d8f" />

This is OpenGL in the appimage from 3bf16fe. Artix Linux.

**Comment by CorySanin** [comment]
Looks fixed to me, I believe #25883 made the difference.
